services/blockstorage/sync/factory.go

```diff
@@ -2,24 +2,79 @@ package sync
 
 import (
 	"github.com/orbs-network/orbs-network-go/instrumentation/log"
+	"github.com/orbs-network/orbs-network-go/instrumentation/metric"
 	"github.com/orbs-network/orbs-spec/types/go/primitives"
 	"github.com/orbs-network/orbs-spec/types/go/protocol/gossipmessages"
 	"github.com/orbs-network/orbs-spec/types/go/services/gossiptopics"
+	"time"
 )
 
 type stateFactory struct {
 	config  blockSyncConfig
 	gossip  gossiptopics.BlockSync
 	storage BlockSyncStorage
+	c       *blockSyncConduit
 	logger  log.BasicLogger
+	metrics *stateMetrics
 }
 
-func NewStateFactory(config blockSyncConfig, gossip gossiptopics.BlockSync, storage BlockSyncStorage, logger log.BasicLogger) *stateFactory {
+type stateMetrics struct {
+	carStateLatency           *metric.Histogram
+	finishedCarStateLatency   *metric.Histogram
+	waitingChunksStateLatency *metric.Histogram
+	idleStateMetrics
+	processingStateMetrics
+}
+
+type idleStateMetrics struct {
+	stateLatency *metric.Histogram
+	timesReset   *metric.Gauge
+	timesExpired *metric.Gauge
+}
+
+type processingStateMetrics struct {
+	stateLatency           *metric.Histogram
+	throughputRate         *metric.Rate
```

Comment:
Rate in what? bytes? blocks?

Reply:
blocks, so originally it was "block-per-second" but that seems wrong as time is probably controlled by the system viewing the data, so i changed to another not-so-good-name, i guess i should call it blocksRate